be/src/vec/exec/scan/new_olap_scanner.cpp

```diff
@@ -481,6 +481,12 @@ Status NewOlapScanner::_init_return_columns() {
         _return_columns.push_back(index);
         if (slot->is_nullable() && !_tablet_schema->column(index).is_nullable()) {
             _tablet_columns_convert_to_null_set.emplace(index);
+        } else if (!slot->is_nullable() && _tablet_schema->column(index).is_nullable()) {
+            throw doris::Exception(ErrorCode::INVALID_SCHEMA,
+                                   "slot(id: {}, name: {})'s nullable does not match "
```

Comment:
could return error status

Reply:
And please check the full call stack to make sure the return status is no discard